### ISSUE TITLE
[Feature]: non-parametric Layers

### ISSUE THREAD

**Issue by lucia-ferrer** [EricssonResearch/illia#95]
### Proposal Type

New Feature

### Proposal Summary

CSWRAPP-85422

Incorporate common non-parametric layers, so as to have a more agnostic definition or API for the modules. 

- [x] Torch
- [ ] Jax
- [x] Tensorflow

### Motivation & Use Case

Non-parametric layers improve illia by usability, and concept-wise defintion:

1. More Complete Neural Architecture Support: Users can build full end-to-end models mixing Bayesian layers (for uncertainty) with standard layers (pooling, activations, normalization) - not just isolated Bayesian components.

2. Backend Consistency: Single API (illia.nn.ReLU, illia.nn.MaxPool2d) works across PyTorch, TensorFlow, and JAX. Users don't need to remember torch.nn vs keras.layers syntax differences.

3. Seamless Integration: Bayesian layers naturally compose with non-parametric layers in the same model:

model = Sequential([
    Conv2d(...),           # Bayesian - uncertainty
    ReLU(),                # Non-parametric - activation
    MaxPool2d(...),        # Non-parametric - downsampling
    BatchNorm2d(...),      # Non-parametric - normalization
    Linear(...)            # Bayesian - uncertainty
])

4. Practical Usability: Without non-parametric layers, users would need to mix illia.nn and native backend imports, breaking the unified API promise and making backend switching harder (even if this is not yet implemented, the goal is there). 

### Technical Details

**Proposed API**: create a wrapper over the native layers, without creating specific new Python classes or modules. This way it simplifies the definition, mantainance, and compared with other alternatives looks the one with smaller overhead. 

### Implementation Status

- [x] Reference implementation exists and is open source
- [ ] Reference implementation exists but is proprietary
- [ ] No existing implementation, but detailed specification available
- [ ] Concept stage - no implementation yet

### Your Contribution Level

I can implement this feature and submit a PR

### Checklist

- [x] I have searched for similar feature requests
- [x] I have provided clear motivation and use cases
- [x] I have included relevant references (if available)
- [x] I understand this is a suggestion and implementation depends on maintainer priorities
